### ISSUE TITLE
Deprecation warnings with Gradle 6.3

### ISSUE THREAD

**Issue by hirakida** [vojtechhabarta/typescript-generator#469]
When I use the gradle plugin with Gradle 6, I get the following deprecation warning.
```
> Task :generateTypeScript
Running TypeScriptGenerator version 2.21.588
The compile configuration has been deprecated for resolution. This will fail with an error in Gradle 7.0. Please resolve the compileClasspath configuration instead. Consult the upgrading guide for further information: https://docs.gradle.org/6.3/userguide/upgrading_version_5.html#dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations
```

**Comment by olegshtch** [comment]
I suppose `implementation` configuration should be enough to generate TS: https://docs.gradle.org/6.3/userguide/java_library_plugin.html#sec:java_library_configurations_graph

@vojtechhabarta
Wonder what to do with deprecation of `compile`? Should we search for `implementation` configuration, and then don't try to resolve `compile`?

**Comment by vojtechhabarta** [comment]
@hirakida thanks for reporting this warning.
@olegshtch thanks for the PR.
I would wait with solution untill Gradle 7 is released because needed change breaks some builds.

**Comment by vojtechhabarta** [comment]
I updated minimal Gradle version to 4, resolved the warning and also updated `sample-gradle` module.

**Comment by vojtechhabarta** [comment]
Released in [2.29.814](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.29.814).
